XLS-0033d-CFTs/README.md

```diff
@@ -161,11 +161,11 @@ Identifies the page in the owner's directory where this item is referenced.
  
 ##### 1.2.1.1.3. How do **`CFTokenIssuance`** objects work?
 
-Any account may issue up to 32 Compact Fungible Tokens, but each issuance must have a different **`AssetCode`**.
+Any account may issue any number of Compact Fungible Tokens.
 
 ###### 1.2.1.1.3.1. Searching for a **`CFTokenIssuance`** object
 
-CFT Issuances are uniquely identified by a combination of a type-specific prefix, the isser address and an asset code. To locate a specific **`CFTokenIssuance`**, the first step is to locate the owner directory for the issuer. Then, find the directory that holds `CFTokenIssuance` ledger objects and iterate through each entry to find the instance with the desired **`AssetCode`**. If that entry does not exist then the **`CFTokenIssuance`** does not exist for the given account.
+CFT Issuances are uniquely identified by a combination of a type-specific prefix, the issuer address and a transaction sequence number. To locate a specific **`CFTokenIssuance`**, the first step is to locate the owner directory for the issuer. Then, find the directory that holds `CFTokenIssuance` ledger objects and iterate through each entry to find the instance with the desired **`AssetCode`**. If that entry does not exist then the **`CFTokenIssuance`** does not exist for the given account.
```

Comment:
With this new proposal, AssetCode will become only metadata, which may or may not used. So now, the `Amount` field will need a change, as opposed to the original design: 
```
"amount": {
  "issuer": "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
  "currency": "USD",
  "value": "1"
}
```
It will now change to 
```
"amount": {
  "cft_issuance_id": "00070C4495F14B0E44F78A264E41713C64B5F89242540EE255534400000000000000",
  "value": "1"
}
```

After specifying the `CFTokenIssuanceID` in a `Payment` transaction, we are still able to fetch the `CFTokenIssuance` in constant time. 

Please let me know if I misunderstood anything

Reply:
How is `cft_issuance_id` obtained?

Reply:
Since the `CFTokenIssuanceID` will be the identifier, whoever wants to make a payment with that specific CFT needs to know the `CFTokenIssuanceID` before hand.

Reply:
Yes, so how do they get this id? It's clear if I want to send say `USD`, but this is just pretty random identifier, which as I user I have to somehow locate.

Reply:
This would mean that we need to add an extra variable in the `Issue` class to encompass for the `CFTokenIssuanceID` in addition to  the `currency` and `issuer`

Reply:
> Yes, so how do they get this id? It's clear if I want to send say USD, but this is just pretty random identifier, which as I user I have to somehow locate.

Yes, USD will no longer be an identifier. You, as the user, needs to keep track of the CFTokenIssuanceID, which is a random identifier. 

The reason why we changed to this approach because there is a problem where the issuer can re-issue the same asset with the same currency after it has been destroyed, which is a problem  for its use cases, according to product.

Reply:
This is why we are using the transaction sequence, which is a non-decreasing index that will uniquely identify a CFT

Reply:
You are talking about implementation and it's not obvious why this should be added to `Issue` since the `currency` is irrelevant in this case. 
My question is how would a client application know the `CFTokenIssuanceID`?

Reply:
Oh, so we are completely relying on clio for API support. 

There are three APIs:

`cfts_by_issuer`: lists all the CFTokenIssuanceID issued by the account
`account_cfts`: all the CFT balances of  an account
`cft_holders`: All the accounts who hold this particular

Reply:
I'm talking about the payment transaction and looking forward offer create, amm create, etc.

Reply:
But it's definitely less convinient for the user since they can no longer specify the currency directly

Reply:
Yes, it is really inconvenient. Is there a PRD or other relevant documentation, which discusses the motivation for this change?

Reply:
I think we can discuss it later in a call.

Reply:
But is there a technical limitation with the new approach? I have done too much with payment or offer so i dont really know.

Reply:
I have not had time to give it a lot of thought. But it seems like on the surface it's a big change.